core/tests/test_utils.py

```diff
@@ -2282,6 +2375,74 @@ def assert_same_list_elements(self, phrases, stdout):
 GenericTestBase = AppEngineTestBase
 
 
+class GenericEmailTestBase(GenericTestBase):
+    """Base class for tests requiring email services."""
+
+    def run(self, result=None):
+        """Adds a context swap on top of the test_utils.run() method so that
+        test classes extending GenericEmailTestBase will automatically have
+        a mailgun api key, mailgun domain name and mocked version of
+        send_email_to_recipients().
+        """
+        with self.swap(feconf, 'MAILGUN_API_KEY', 'key'), (
+            self.swap(feconf, 'MAILGUN_DOMAIN_NAME', 'name')), (
+                self.swap(
+                    email_services, 'send_email_to_recipients',
+                    self.email_services_mock.mock_send_email_to_recipients)):
+            super(EmailTestBase, self).run(result=result)
+
+    @classmethod
+    def setUpClass(cls):
+        super(GenericEmailTestBase, cls).setUpClass()
+        cls.email_services_mock = EmailServicesMock()
+
+    def setUp(self):
+        super(GenericEmailTestBase, self).setUp()
+        self.email_services_mock.wipe_emails_dict()
+
+
+EmailTestBase = GenericEmailTestBase
+
+
+class MockLoggingHandler(python_utils.OBJECT):
```

Comment:
Use same approach as mentioned above.

Basically the issue with the current approach is that a LoggingHandler isn't a thing. When you declare a mock, it should be the same type as an existing thing so that you can swap it out cleanly. In this case it looks like you're swapping at the method level, so just make sure the methods are available in your subclass.

Additionally, since the logging thing isn't related to emails, that should be part of the specific test class rather than the general infrastructure here.